search/searcher/search_term.go

```diff
@@ -42,7 +42,7 @@ func NewTermSearcher(indexReader index.IndexReader, term string, field string, b
 }
 
 func NewTermSearcherBytes(indexReader index.IndexReader, term []byte, field string, boost float64, options search.SearcherOptions) (*TermSearcher, error) {
-	needFreqNorm := options.Score != "none"
+	needFreqNorm := options.IncludeTermVectors || options.Score != "none"
 	reader, err := indexReader.TermFieldReader(term, field, needFreqNorm, needFreqNorm, options.IncludeTermVectors)
```

Comment:
Instead of checking this at this level, I'm thinking the indexReader.TermFieldReader() has all the information that it needs to check this directly and more exactly?  That is, indexReader.TermFieldReader() also has the options.IncludeTermVectors bool? 

In contrast, the proposal here kinda overloads the `needFreqNorm` flag with "hidden" information about IncludeTermVectors?

Reply:
Ya true, but with the code today, we initialize freqNormReader only when needFreqNorm is set to true, and this is needed to check and see if locations are actually available ..
https://github.com/blevesearch/bleve/blob/7f5d67a851143ce1ad4d8a0d493362300a814ccc/index/scorch/segment/zap/posting.go#L420-L425

And it became a somewhat invasive change to accommodate the situation where locations are required when score isn't - where now I'd need to check includeFreqNorm || includeLocs, and I felt more comfortable with this change at this stage.